cmd/server/server.go

```diff
@@ -143,6 +142,13 @@ func (s *server) startService() common.Daemon {
 		params.MessagingClient = nil
 	}
 
+	if params.ClusterMetadata.IsArchivalEnabled() {
```

Comment:
any reason this lives inside cluster metadata instead of being a flag like s.cfg.Archival.Enabled ? My understanding is cluster metadata is reserved for things like cluster name / addr / peers etc and not for other server side feature flags

Reply:
Talked offline about this one.